### ISSUE TITLE
Regarding reproducability of log loss

### ISSUE THREAD

**Issue by Kshitij68** [mljar/mljar-supervised#565]
Hi, 

I was trying to reproduce the log loss calculation and found that the values do not match when the training and prediction set are the same.
Is it expected or it is a buggy behavior?
```
        model = AutoML(
            explain_level=0, verbose=1, random_state=1, results_path=self.automl_dir
        )
        model.fit(iris.data, iris.target)
        predictions = model.predict_proba(iris.data)
        params = json.load(open(os.path.join(self.automl_dir, "params.json")))
        leaderboard = pd.read_csv(os.path.join(self.automl_dir, "leaderboard.csv"))
        assert params['eval_metric'] == 'logloss'
        metric = Metric({"name": "logloss"})
        prediction_log_loss = metric(y_true=iris.target, y_predicted=predictions)
        training_log_loss = leaderboard[leaderboard['name'] == params['best_model']]['metric_value'].values[0]
        assert round(training_log_loss,3) == round(prediction_log_loss,3)
```

**Comment by pplonski** [comment]
Metrics reported in AutoML are computed based on validation data. In your case, you have used train/test split.

1. Data is is splitted into two sets. The 75% of all samples are train data. The 25% of all samples are validation data.
2. The logloss in the leaderboard is computed on 25% of the data (validation samples).
